packages/marko-web-omeda-identity-x/middleware/set-promo-source.js

```diff
@@ -0,0 +1,36 @@
+const clean = (value) => {
+  if (!value || value === 'null') return null;
+  return value.replace(/"'/g, '').trim().toUpperCase();
+};
+
+/**
+ * Sets the `oly_promo_src` cookie to identify the customer's original promotion source if the
+ * `oly_promo_src` URL parameter is present.
+ * '
+ * @param {String} omedaPromoCodeCookieName The name of the cookie to interact with
+ * @param {Boolean} stripIncomingPromoCode
+ */
+module.exports = ({
+  omedaPromoCodeCookieName = 'oly_promo_src',
+  stripIncomingPromoCode = true,
+} = {}) => (req, res, next) => {
+  const promoSource = clean(req.cookies[omedaPromoCodeCookieName]);
+
+  const { [omedaPromoCodeCookieName]: promoCode, ...q } = req.query;
+  const incomingPromoCode = clean(promoCode);
+
+  // Only set the promo source cookie if it doesn't already exist
+  if (!promoSource && incomingPromoCode) {
```

Comment:
I disagree here... the incoming param should override the current cookie value, otherwise new campaign source attribution will be lost